extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -371,37 +371,45 @@ export class AmpAnalytics extends AMP.BaseElement {
    * @private
    */
   handleEvent_(trigger, event) {
-    let request = this.requests_[trigger['request']];
-    if (!request) {
-      user().error(this.getName_(), 'Ignoring event. Request string ' +
-          'not found: ', trigger['request']);
-      return Promise.resolve();
-    }
-
-    // Add any given extraUrlParams as query string param
-    if (this.config_['extraUrlParams'] || trigger['extraUrlParams']) {
-      const params = Object.create(null);
-      Object.assign(params, this.config_['extraUrlParams'],
-          trigger['extraUrlParams']);
-      for (const k in params) {
-        params[k] = this.expandTemplate_(params[k], trigger, event);
+    const requests = Array.isArray(trigger['request'])
+        ? trigger['request'] : [trigger['request']];
+
+    const resultPromises = [];
```

Comment:
Generally, I think it'd be the best to move the body of the cycle to a separate function. The Git diff will be much smaller as well in this case and harder to make accidental return errors.